src/Analyzers/CSharp/Analyzers/UseCollectionInitializer/CSharpUseCollectionInitializerAnalyzer.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.UseCollectionInitializer;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseCollectionInitializer;
+
+internal sealed class CSharpUseCollectionInitializerAnalyzer : AbstractUseCollectionInitializerAnalyzer<
+    ExpressionSyntax,
+        StatementSyntax,
+        BaseObjectCreationExpressionSyntax,
+        MemberAccessExpressionSyntax,
+        InvocationExpressionSyntax,
+        ExpressionStatementSyntax,
+        ForEachStatementSyntax,
+        IfStatementSyntax,
+        VariableDeclaratorSyntax,
+        CSharpUseCollectionInitializerAnalyzer>
+{
+    protected override bool IsComplexElementInitializer(SyntaxNode expression)
+        => expression.IsKind(SyntaxKind.ComplexElementInitializerExpression);
+
+    protected override void GetPartsOfForeachStatement(
+        ForEachStatementSyntax statement,
+        out SyntaxToken identifier,
+        out ExpressionSyntax expression,
+        out IEnumerable<StatementSyntax> statements)
+    {
+        identifier = statement.Identifier;
+        expression = statement.Expression;
+        statements = ExtractEmbeddedStatements(statement.Statement);
+    }
+
+    protected override void GetPartsOfIfStatement(
+        IfStatementSyntax statement,
+        out ExpressionSyntax condition,
+        out IEnumerable<StatementSyntax> whenTrueStatements,
+        out IEnumerable<StatementSyntax>? whenFalseStatements)
```

Comment:
same issue.  C# has `if/else` (Where else is optional).  VB has `if/elif/elif/else`.  These are really different forms, and trying to shoehorn into ISyntaxFacts was not the right way to do things.